hw/ip/i2c/doc/_index.md

```diff
@@ -1,3 +1,225 @@
+These timings are then fed directly to the I2C state machine to control the bus timing.  
+
+### Timeout Control
+A malfunctioning (or otherwise very slow) target device can hold SCL low indefinitely stalling the bus.
+For this reason the {{< regref TIMEOUT_CTRL >}} provide a clock-stretching timeout mechanism to notify firmware of this sort of condition.
+If {{< regref TIMEOUT_CTRL.EN >}} is asserted, an interrupt will be asserted the IP detects that a remote device has been holding SCL low for more than {{< regref TIMEOUT_CTRL.VAL >}} clock ticks.
+
+
+This feature is added as a utility, though it is not required by the I2C specification.
+However, in some applications it could be used in protocols which build upon I2C.
+For instance, SMBus applications using this IP could in principle use this to support SMBus timeouts.
+(Note: This is just an example application of this feature.
+Other features may also be required for complete SMBus functionality.)
+
+
+### Timing Control Registers
+The I2C module has a few interrupts including general data flow interrupts and unexpected event interrupts.
+
+If the RX FIFO exceeds the designated depth of entries, the interrupt `rx_watermark` is raised to inform firmware.
+Firmware can configure the watermark value via the register {{< regref FIFO_CTRL.RXILVL >}}.
+
+Meanwhile it the FMT FIFO level falls below a designated depth of entries the `fmt_watermark` interrupt is raised.
+(Note that this behavior differsfrom similar interrupts in other modules, such as the UART IP module.)
+Firmware can configure the watermark value via the register {{< regref FIFO_CTRL.FMTILVL >}}.
+
+If either FIFO receives an additional write request when its FIFO is full, the interrupt `fmt_overflow` or `rx_overflow` is asserted and the format indicator or character is dropped.
+
+If the module transmits a byte, but receives no ACK signal, the `nak` interrupt is usually asserted.
+In cases where a byte is transmitted and no ACK is expected or required, that byte should be submitted with NAKOK flag also asserted.
+
+When the I2C module is in transmit mode, the `scl_interference` or `sda_interference` interrupts will be asserted if the IP identifies that some other device on the bus is forcing either signal low and interfering with the transmission.
+If should be noted that the `scl_interference` interrupt is not raised in the case when the target device is stretching the clock.
+(However, it may be raised if the target allows SCL to go high and then pulls SCL down before the end of the current clock cycle.)
+
+Though normal clock stretching does not count as SCL interference, if the module detects that a target device has held SCL low and stretched the any given SCL cycle for more than {{< regref TIMEOUT_CTRL.VAL >}} clock ticks this will cause the stretch timeout interrupt to be asserted.
```

Comment:
some wavejson diagrams for clock stretching would be helpful.

Reply:
Good idea.  I've added a couple in commit https://github.com/lowRISC/opentitan/pull/1047/commits/65bc6d7049915fb7f5b63bb5404ddc37b3d0c8c4

(Note will be squashed at the end of this)